portfolio/src/main/java/com/google/sps/servlets/CommentsListServlet.java

```diff
@@ -15,31 +16,57 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime; 
 
-/** Servlet responsible for listing tasks. */
+
+/** Servlet responsible for listing comments. */
 @WebServlet("/comments-list")
 public class CommentsListServlet extends HttpServlet {
-
+  
+  private DatastoreService datastore; 
+  
+  @Override
+  public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  } 
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numComments = getMax(request); 
     Query query = new Query("Comment").addSort("timestamp_millis", SortDirection.DESCENDING);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String content = (String) entity.getProperty("content");
-      long timestamp_millis = (long) entity.getProperty("timestamp_millis");
+    if(numComments < 4){
```

Comment:
Why is numComments >= 4 handled differently from numComments < 4? The implementations look nearly identical, and I don't see any reason why a `results.asList(FetchOptions.Builder.withLimit(numComments)))` can't be used for both cases.

Reply:
In the selection menu, the value 4 is associated with displaying all comments so if the value is less than four only 1,2,or 3 comments are displayed

Reply:
This is weird. It's not unprecedented for some integer to stand in for a non-integer value, but typically this would be `-1` or `0`. -1 sounds better to me here, so if you're going to have a special value for this, consider -1.